### ISSUE TITLE
conflict in views with spack stacks

### ISSUE THREAD

**Issue by lee218llnl** [spack/spack#11708]
I am getting errors like this:

```
==> Error: [/usr/workspace/lee218/delete/usr/tce] Package conflict detected:
   (Linked) fftw@3.3.8%gcc@4.9.3+double+float~fma+long_double+mpi~openmp~pfft_patches~quad simd=avx,avx2,sse2 arch=linux-rhel7-x86_64/rgyy7pr
(Specified) fftw@3.3.8%gcc@4.9.3+double+float~fma+long_double+mpi~openmp~pfft_patches~quad simd=avx,avx2,sse2 arch=linux-rhel7-x86_64/q3bdqd7
```

This is because in my spack.yaml I have fftw^mpi as a combinatorial package and I specify both mvapich2 and openmpi as mpi_specs. Also in the spack.yaml, here is the view specification:

```
  view:
    default:
      root: /usr/workspace/lee218/delete/usr/tce
      select: ['%gcc@4.9.3', intel@18.0.1]
      projections: {}
    tce-llnl:
      root: /usr/workspace/lee218/delete/usr/tce/packages
      projections:
        ^mpi: '{name}/{name}-{version}-{^mpi.name}-{^mpi.version}-{compiler.name}-{compiler.version}'
        '%gcc@4.9.3': '{name}/{name}-{version}'
        all: '{name}/{name}-{version}-{compiler.name}-{compiler.version}'
```

We probably do not want the combinatorial_packages to show up in the default view, unless a specific ^mpi%compiler is explicitly requested.

**Comment by becker33** [comment]
I think the best way to achieve this right now is

```
  view:
    default:
      root: /usr/workspace/lee218/delete/usr/tce
      select: ['%gcc@4.9.3', intel@18.0.1]
      exclude: ['^openmpi']
      projections: {}
```

@lee218llnl

**Comment by alalazo** [comment]
See also the comment in https://github.com/spack/spack/pull/11372#issuecomment-489397577 proposing a possible way to avoid this behavior as default (basically activate views only when you concretize things together).

**Comment by tgamblin** [comment]
@becker33: is this one going to be resolved with any of the 0.13.0 fixes?

**Comment by tgamblin** [comment]
Closing this because it's a configuration issue -- @becker33's solution above should work.
